packages/yarnpkg-core/sources/Plugin.ts

```diff
@@ -73,6 +75,14 @@ export type Hooks = {
     extra: {script: string, args: Array<string>, cwd: PortablePath, env: ProcessEnvironment, stdin: Readable | null, stdout: Writable, stderr: Writable},
   ) => Promise<() => Promise<number>>;
 
+  /**
+   * Called when a network request is being made.
+   */
+  wrapNetworkRequest?: (
+    networkRequest: () => Promise<any>,
+    extra: {target: string | URL, body: Body, configuration: Configuration, headers: { [headerName: string]: string } | undefined, jsonRequest: boolean | undefined, jsonResponse: boolean | undefined, method: Method | undefined}
+  ) => Promise<() => Promise<any>>;
```

Comment:
I'd prefer if this had its own type. Somewhere at the beginning of the file you can do something like:

```ts
export type WrapNetworkRequestInfo = httpUtils.Options & {
  target: string | URL;
  body: Body;
};
```

Then reuse this type in the parameter (and don't forget to export this type from `index.ts`).

Reply:
Done